src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -1250,6 +1257,16 @@ DecideTransactionPropertiesForTaskList(RowModifyLevel modLevel, List *taskList,
 void
 StartDistributedExecution(DistributedExecution *execution)
 {
+	if (!execution->localExecutionSupported)
```

Comment:
I don't think it is necessary to throw error at this point. If `IfTransactionAccessedPlacementsLocally`, we'd not have any local tasks that we'd try to run via remote execution.

`CreateDistributedExecution` extracts the local tasks `if (GetCurrentLocalExecutionStatus() == LOCAL_EXECUTION_REQUIRED)`. It means that there is no need to call `ErrorIfTransactionAccessedPlacementsLocally`.

Am I missing anything? Does removing this change any test result? This seems like an `Assert` that would not be triggered by any tx block (or execution in general) ?

Reply:
Even if you try:
```SQL
 BEGIN;
 select * from test where a = 8;
 a 
---
(0 rows)

 set citus.enable_local_execution to false;
SET
Time: 0.352 ms
 select * from test where a = 8;
ERROR:  cannot switch local execution status from local execution required to local execution disabled since it can cause visibility problems in the current transaction
```

We guard it in `StartPlacementExecutionOnSession->SetLocalExecutionStatus`. So, the TX fails when/if needed for such extreme cases

Reply:
I think you're right, looks like I missed removing that when reverting the previous version of the fix.
Reverted now.